pkg/vm/memory/relocatable.go

```diff
@@ -6,89 +6,162 @@ import (
 	f "github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
 )
 
-// The default segment that represents ordinary felts.
-const feltSegment = 0
-
 // Represents a Memory Address of the Cairo VM. Because memory is split between different segments
 // during execution, addresses has two locators: the segment they belong to and the location
 // inside that segment
 type MemoryAddress struct {
 	SegmentIndex uint64
-	Offset       f.Element
+	Offset       *f.Element
 }
 
 // Creates a new memory address
 func CreateMemoryAddress(segment uint64, offset *f.Element) *MemoryAddress {
-	return &MemoryAddress{SegmentIndex: segment, Offset: *offset}
+	return &MemoryAddress{SegmentIndex: segment, Offset: offset}
 }
 
-// Adds two relocatables if the second one is a Felt (relocatable of the form 0:off).
-func (r *MemoryAddress) Add(r1 *MemoryAddress, r2 *MemoryAddress) (*MemoryAddress, error) {
-	if r2.SegmentIndex != feltSegment {
-		return nil, fmt.Errorf("cannot add two relocatables")
-	}
+func (address MemoryAddress) String() string {
+	return fmt.Sprintf("Memory Address: segment: %d, offset: %s", address.SegmentIndex, address.Offset.Text(10))
```

Comment:
A todo for the future, we might need a nice printing function that output -1 instead of (PRIME - 1), fore example